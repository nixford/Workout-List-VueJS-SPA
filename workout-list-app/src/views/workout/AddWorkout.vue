<template>
    <form id="add">
        <div class="input-row">
          <label>Workout type:</label>
          <input type="text" v-model="workoutType">
        </div>  
        <div class="add-exercise-btn">
          <h3>Add exercises:</h3>     
        </div>        
        <div class="input-row-exercises"> 
          <div>
            <label>Exercise name:</label>      
            <input type="text" v-model="excersiceName">
          </div>   
          <div>
            <label>Repets count:</label>
            <input type="number" v-model="repeats">
          </div>
          <div> 
            <label>Sets count:</label>
            <input type="number" v-model="sets">
          </div>        
        </div>
        <div class="exercises-div" v-for="(e, i) in exercises" :key="e.id"> 
          <p>
            <span class="exercises-count">{{i + 1}}. </span>
            <span>Exercise name: {{e.excersiceName}} </span>
            <span> - </span>
            <span>Repets count: {{e.repeats}} </span>
            <span> / </span>
            <span>Sets count: {{e.sets}} </span>
          </p>
        </div>
        <button submit.prevent @click="addExercise" class="btn-add-exercise">ADD EXERCISE</button> 
    </form>
    <button @click="save" class="btn-save">SAVE WORKOUT</button> 
</template>

<script>
import uniqueId from 'lodash/uniqueId';
export default {
  data(){
    return {
      workoutType: '',
      exercises: [],
      excersiceName: '',
      repeats: null,
      sets: null,
    }
  },
  methods: {
    addExercise(e) {
      e.preventDefault();
      let exercise = {
        id: uniqueId(),
        excersiceName: this.excersiceName,
        repeats: this.repeats,
        sets: this.sets,
        isCompleted: false,
      }
      this.exercises.push(exercise);
      exercise = {};
      this.excersiceName = '';
      this.repeats = null;
      this.sets = null;
    },
    save() {
      let workout = {
        id: uniqueId(),
        workoutType: this.workoutType,
        exercises: this.exercises,
      }
      this.$store.commit('addToList', workout);
      this.$router.push('/');
    }
  },  
}
</script>

<style lang="scss" scoped>
#add {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 3%;
}

.input-row {
  display: flex;
  flex-direction: column;
  width: 60%;
  margin-top: 3%;
}

input {
  height: 25px;
}

.add-exercise-btn {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin: 3% 3% 3% 3%;

  button {
    margin-left: 30px;
  }
}

.btn-add-exercise {
  width: 180px;
  height: 30px;
  border-radius: 12px;
  background-color: orange;
  color: white;
  font-weight: 700;
  margin-top: 3%;
}

.input-row-exercises {
  display: flex;
  justify-content: space-evenly;
  flex-wrap: wrap;
  width: 80%;
}

.btn-save {
  margin-top: 5%;
  width: 50%;
  height: 50px;
  border-radius: 12px;
  background-color: green;
  color: white;
  font-weight: 700;
}
</style>