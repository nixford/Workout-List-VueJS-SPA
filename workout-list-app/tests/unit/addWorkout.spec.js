import { shallowMount } from '@vue/test-utils';
import AddWorkout from '../../src/views/workout/AddWorkout.vue';

describe('AddWorkout.vue', () => {

  const wrapper = shallowMount(AddWorkout); 

  test('initial length is zero', () => { 
    // Act
    expect(wrapper.vm.$data.exercises.length).toBe(0);
  })

  test('exercise is added to array when add button is clicked', () => {    
    // Arrange
    const button = wrapper.find('button');

    // Act
    button.trigger('click');

    // Assert
    expect(wrapper.vm.$data.exercises.length).toBe(1);
  })

  test('exercise is listed on the page when add button is clicked', async () => {  
    // Arrange  
    const button = wrapper.find('button');

    // Act
    await button.trigger('click');

    // Assert
    expect(wrapper.find('p').text()).toContain("1.");
  })

  test('snapshot renders correctly', () => {  
    expect(wrapper.element).toMatchSnapshot();
  })
  
  test('snapshot contains h1 tag', () => {    
    expect('<h1>Workout List App</h1>').toMatchSnapshot();
  })
})
