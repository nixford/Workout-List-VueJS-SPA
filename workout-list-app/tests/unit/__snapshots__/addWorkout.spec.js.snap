// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWorkout.vue snapshot contains h1 tag 1`] = `<h1>Workout List App</h1>`;

exports[`AddWorkout.vue snapshot renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <form
    id="add"
  >
    <div
      class="input-row"
    >
      <label>
        Workout type:
      </label>
      <input
        type="text"
      />
    </div>
    <div
      class="add-exercise-btn"
    >
      <h3>
        Add exercises:
      </h3>
    </div>
    <div
      class="input-row-exercises"
    >
      <div>
        <label>
          Exercise name:
        </label>
        <input
          type="text"
        />
      </div>
      <div>
        <label>
          Repets count:
        </label>
        <input
          type="number"
        />
      </div>
      <div>
        <label>
          Sets count:
        </label>
        <input
          type="number"
        />
      </div>
    </div>
    
    <div
      class="exercises-div"
    >
      <p>
        <span
          class="exercises-count"
        >
          1. 
        </span>
        <span>
          Exercise name: 
        </span>
        <span>
           - 
        </span>
        <span>
          Repets count: 
        </span>
        <span>
           / 
        </span>
        <span>
          Sets count: 
        </span>
      </p>
    </div>
    <div
      class="exercises-div"
    >
      <p>
        <span
          class="exercises-count"
        >
          2. 
        </span>
        <span>
          Exercise name: 
        </span>
        <span>
           - 
        </span>
        <span>
          Repets count: 
        </span>
        <span>
           / 
        </span>
        <span>
          Sets count: 
        </span>
      </p>
    </div>
    
    <button
      class="btn-add-exercise"
      submit.prevent=""
    >
      ADD EXERCISE
    </button>
  </form>
  <button
    class="btn-save"
  >
    SAVE WORKOUT
  </button>
  
</div>
`;
